src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotPassTypesByReferenceTests.cs

```diff
@@ -0,0 +1,171 @@
+    Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.DoNotPassTypesByReference,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.UnitTests
+{
+    public class DoNotPassTypesByReferenceTests
+    {
+        [Fact]
+        public async Task CA1045_RefParameters_Diagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public class Class1
+{
+    public void Method1(ref string s, ref object o)
+    {
+    }
+}",
+                GetCA1045CSharpResultAt(4, 36, "s"),
+                GetCA1045CSharpResultAt(4, 50, "o"));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Class Class1
+    Public Sub Method1(ByRef s As String, ByRef o As Object)
+    End Sub
+End Class",
+                GetCA1045BasicResultAt(3, 30, "s"),
+                GetCA1045BasicResultAt(3, 49, "o"));
+        }
+
+        [Fact]
+        public async Task CA1045_MethodIsNotPublic_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public class Class1
+{
+    private void Method1(ref string s)
+    {
+    }
+}");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Class Class1
+    Private Sub Method1(ByRef s As String)
+    End Sub
+End Class");
+        }
+
+        [Fact]
+        public async Task CA1045_MethodIsOverride_Diagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public class BaseClass
+{
+    public virtual void Method1(ref string s) // issue here...
+    {
+    }
+}
+
+public class Class1 : BaseClass
+{
+    public override void Method1(ref string s) // ... but not here
+    {
+    }
+}",
+                GetCA1045CSharpResultAt(4, 44, "s"));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Class BaseClass
+    Public Overridable Sub Method1(ByRef s As String) ' issue here...
+    End Sub
+End Class
+
+Public Class Class1
+    Inherits BaseClass
+
+    Public Overrides Sub Method1(ByRef s As String) ' ... but not here
+    End Sub
+End Class
+",
+                GetCA1045BasicResultAt(3, 42, "s"));
+        }
+
+        [Fact]
+        public async Task CA1045_MethodIsInterfaceImplementation_Diagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public interface Interface1
+{
+    void Method1(ref string s); // issue here...
+}
+
+public class Class1 : Interface1
+{
+    public void Method1(ref string s) // ... but not here
+    {
+    }
+}",
+                GetCA1045CSharpResultAt(4, 29, "s"));
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Interface Interface1
+    Sub Method1(ByRef s As String) ' issue here...
+End Interface
+
+Public Class Class1
+    Implements Interface1
+
+    Public Sub Method1(ByRef s As String) Implements Interface1.Method1 ' ... but not here
+    End Sub
+End Class",
+                GetCA1045BasicResultAt(3, 23, "s"));
+        }
+
+        [Fact]
+        public async Task CA1045_VBNET_ByRefOutParameter_NoDiagnostic()
```

Comment:
💡 Consider updating this test to use methods with one `ref` parameter and one `out` parameter, so it's clear that the distinguishing characteristic of the `out` parameter is the use of `out`, as opposed to an unintentional situation like limited visibility of the method causing the analyzer to ignore it.